include/NeoFOAM/core/demangle.hpp

```diff
@@ -19,7 +19,7 @@ template<typename T, typename Container, typename Key>
 void logBadAnyCast(const std::bad_any_cast& e, const Key& key, const Container& data)
 {
     std::cerr << "Caught a bad_any_cast exception: \n"
-              << "requested type " << demangle(typeid(T).name()) << "\n"
+              << "key requested: " << key << "requested type " << demangle(typeid(T).name()) << "\n"
               << "actual type " << demangle(data.at(key).type().name()) << "\n"
               << e.what() << std::endl;
```

Comment:
would https://en.cppreference.com/w/cpp/utility/format/format work? Could look like (not tested):

```suggestion
              << std::format("Caught a bad_any_cast exception: key requested: {}, requested type: {}, actual type: {}\n" , 
              key, 
              demangle(typeid(T).name()),
              demangle(data.at(key).type().name()))
```